api/pipeline_spec.proto

```diff
@@ -0,0 +1,420 @@
+      // matches with the `output_parameter_key`.
+      string producer_task = 1;
+
+      // The key of [TaskOutputsSpec.parameters][] map of the producer task.
+      string output_parameter_key = 2;
+    }
+    oneof kind {
+      // Output parameter from an upstream task.
+      TaskOutputParameterSpec task_output_parameter = 1;
+      // A constant value or runtime parameter.
+      ValueOrRuntimeParameter runtime_value = 2;
+    }
+  }
+
+  // A map of input parameters which are small values, stored by the system and
+  // can be queriable.
+  map<string, InputParameterSpec> parameters = 1;
+  // A map of input artifacts.
+  map<string, InputArtifactSpec> artifacts = 2;
+}
+
+// The spec of task outputs.
+message TaskOutputsSpec {
+  // The specification of a task output artifact.
+  message OutputArtifactSpec {
+    // The type of the artifact.
+    ArtifactTypeSchema artifact_type = 1;
+
+    // The properties of the artifact, which are determined either at
+    // compile-time, or at pipeline submission time through runtime parameters
+    map<string, ValueOrRuntimeParameter> properties = 2;
+
+    // The custom properties of the artifact, which are determined either at
+    // compile-time, or at pipeline submission time through runtime parameters
+    map<string, ValueOrRuntimeParameter> custom_properties = 3;
+  }
+
+  // Specification for output parameters produced by the task.
+  message OutputParameterSpec {
+    // Required field. The type of the output parameter.
+    PrimitiveType.PrimitiveTypeEnum type = 1;
+  }
+
+  // A map of output parameters which are small values, stored by the system and
+  // can be queriable. The output key is used
+  // by [TaskInputsSpec.InputParameterSpec][] of the downstream task to specify
+  // the data dependency. The same key will also be used by
+  // [ExecutorInput.Inputs][] to reference the output parameter.
+  map<string, OutputParameterSpec> parameters = 1;
+  // A map of output artifacts. Keyed by output key. The output key is used
+  // by [TaskInputsSpec.InputArtifactSpec][] of the downstream task to specify
+  // the data dependency. The same key will also be used by
+  // [ExecutorInput.Inputs][] to reference the output artifact.
+  map<string, OutputArtifactSpec> artifacts = 2;
+}
+
+// Represent primitive types. The wrapper is needed to give a namespace of
+// enum value so we don't need add `PRIMITIVE_TYPE_` prefix of each enum value.
+message PrimitiveType {
+  // The primitive types.
+  enum PrimitiveTypeEnum {
+    PRIMITIVE_TYPE_UNSPECIFIED = 0;
+    INT = 1;
+    DOUBLE = 2;
+    STRING = 3;
+  }
+}
+
+// The spec of a pipeline task.
+message PipelineTaskSpec {
+  // Basic info of a pipeline task.
+  PipelineTaskInfo task_info = 1;
+
+  // Specification for task inputs which contains parameters and artifacts.
+  TaskInputsSpec inputs = 2;
+
+  // Specification for task outputs.
+  TaskOutputsSpec outputs = 3;
+
+  // Label for the executor of the task.

```

Comment:
What is this label used for?

Reply:
It's the key to be referenced in the deployment config.